### ISSUE TITLE
Question about the coredns-mdns use-case within k8s

### ISSUE THREAD

**Issue by fanminshi** [openshift/coredns-mdns#75]
Hi,

I would like to understand how is coredns-mdns be used in context of k8s. specifically what problems it tries to solve?
To give a context, I am trying to figure how to resolve the api-server via mdns-name for binary such as kubelet and kube-proxy. 

Thanks,
Fanmin

**Comment by cybertron** [comment]
Sorry for the delay responding.

Our use of coredns-mdns was to provide DNS records for the nodes in the cluster early in the deployment process, even before the node has joined the cluster. We actually stopped using it a while back because our deployment process changed so we no longer needed DNS records that early, so we now populate those records using the node list from the Kubernetes API. The nice thing about using the API is that we no longer need multicast traffic so it scales much better.

We never used mdns directly for looking anything up. We used this coredns plugin to translate the mdns records being published into regular DNS records, and then lookups were done against that DNS server. It was just a mechanism to generate DNS records.
